plugins/context_datalayer_reaction.inc

```diff
@@ -162,11 +162,14 @@ class context_datalayer_reaction extends context_reaction {
     foreach ($this->get_contexts() as $context) {
       $options = $this->fetch_from_context($context);
       if (!empty($options['data'])) {
+        $data = array_map(function($value) {
+          return token_replace($value);
+        }, $options['data']);
```

Comment:
I think this is fine. The assumption here is that the user is entering simple strings or tokens into the reaction form so the node considerations, I feel, are probably outside of scope.